skywalking/trace/context.py

```diff
@@ -150,22 +151,24 @@ def ignore_check(self, op: str, kind: Kind):
         return None
 
     def start(self, span: Span):
+        self.nspans += 1
         spans = _spans()
         if span not in spans:
             spans.append(span)
 
     def stop(self, span: Span) -> bool:
+        self.nspans -= 1
```

Comment:
move to line 165?

Reply:
Not really necessary because at this point `finish()` always actually finishes since I changed the depth check in `StackedSpan` from `finish()` to `stop()` where it more naturally belongs, so doesn't really matter where it goes and the `if span.finish(self.segment)` will always succeed. In fact I should maybe get rid of it?